### PR TITLE
New react docs (only content)

### PR DESCRIPTION
**Pull request by ryyppy**
I created this branch as a replacement for #60 to be able to iterate on the React content independently from the website infrastructure. Turned out writing content for React is a long process, and I still want to work on other parts of the website :-)

Within this long running PR, only touch following files:

- pages/docs/react/latest
- scripts/extract-tocs.js

### CODE REVIEW

File: pages/docs/react/latest/hooks-overview.mdx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+<!-- - [useCallback](./hooks-callback): Returns a memoized callback -->
+<!-- - [useMemo](./hooks-memo): Returns a memoized value -->
+- [useRef](./hooks-ref): Returns a mutable React-Ref value
+<!-- - [useImperativeHandle](./hooks-imperative-handle): Customizes the instance value that is exposed to parent components when using `ref` -->
+<!-- - [useLayoutEffect](./hooks-layout-effect): Identical to useEffect, but it fires synchronously after all DOM mutations. -->
+
+
+## Rules of Hooks
+
+Hooks are just simple functions, but you need to follow _two rules_ when using them. ReScript doesn't enforce those rules within the compiler, so if you really want to enforce correct hooks conventions, you can use an [eslint-plugin](https://www.npmjs.com/package/eslint-plugin-react-hooks) to check your compiled JS output.
```
Comment (thezjy) [inline pages/docs/react/latest/hooks-overview.mdx:97]: I'm just doing this (eslint the compiled JS output) and wondering whether it's the right way!

**Comment by mweichert** [comment]
Excellent work thus far!

**Comment by bdunn313** [comment]
@ryyppy is there an opportunity to assist with this task? I'd love to point my team at one place for all our rescript / react docs, so I'd love to roll up my sleeves and assist.

It's a little unclear to me what the outstanding tasks are, if you could briefly point me in the right direction so I'm not stepping on your toes, I'll get to work! :D

**Comment by mweichert** [comment]
Thanks for all the work on this @ryyppy.

> ReScript JSX and some rescript-react API changes

Is there a link that references the changes you're referring to above?

**Comment by ryyppy** [comment]
> Is there a link that references the changes you're referring to above?

Not yet, Ricky and I are currently writing things up so we can discuss it with the community to address any edge cases for an easy migration

**Comment by ryyppy** [comment]
I will merge this now to get the ball rolling

**Comment by bdunn313** [comment]
This is a great birthday present (31st)! It begins! 🥳

**Pull request merged**